algorithms/kernel/kernel_function/kernel_function_rbf_dense_default_impl.i

```diff
@@ -133,190 +166,105 @@ services::Status KernelImplRBF<defaultDense, algorithmFPType, cpu>::computeInter
+    daal::threader_for(nBlocks1, nBlocks1, [&](const size_t iBlock1) {
+        DAAL_INT nRowsInBlock1 = (iBlock1 != nBlocks1 - 1) ? blockSize : nVectors1 - iBlock1 * blockSize;
+        DAAL_INT startRow1     = iBlock1 * blockSize;
+        DAAL_INT endRow1       = startRow1 + nRowsInBlock1;
 
-        daal::threader_for(blockNum1, blockNum1, [&](const size_t iBlock) {
-            const size_t numRowsInBlock = (iBlock != blockNum1 - 1) ? blockSize : nVectors1 - iBlock * blockSize;
-            const size_t startRow       = iBlock * blockSize;
-            const size_t finishRow      = startRow + numRowsInBlock;
+        ReadRows<algorithmFPType, cpu> mtA1(*const_cast<NumericTable *>(a1), startRow1, nRowsInBlock1);
+        DAAL_CHECK_BLOCK_STATUS_THR(mtA1);
+        const algorithmFPType * dataA1 = const_cast<algorithmFPType *>(mtA1.get());
 
-            ReadRows<algorithmFPType, cpu> mtA1(*const_cast<NumericTable *>(a1), startRow, numRowsInBlock);
-            DAAL_CHECK_BLOCK_STATUS_THR(mtA1);
-            const algorithmFPType * dataA1 = const_cast<algorithmFPType *>(mtA1.get());
+        WriteOnlyRows<algorithmFPType, cpu> mtR(r, startRow1, nRowsInBlock1);
+        DAAL_CHECK_BLOCK_STATUS_THR(mtR);
+        algorithmFPType * dataR = mtR.get();
 
-            for (size_t i = startRow, ii = 0; i < finishRow; ++i, ++ii)
-            {
-                sqrDataA1[i] = zero;
-                for (size_t j = 0; j < nFeatures; ++j)
-                {
-                    sqrDataA1[i] += dataA1[ii * nFeatures + j] * dataA1[ii * nFeatures + j];
-                }
-            }
-        });
+        daal::threader_for(nBlocks2, nBlocks2, [&, nVectors2, nBlocks2](const size_t iBlock2) {
+            DAAL_INT nRowsInBlock2 = (iBlock2 != nBlocks2 - 1) ? blockSize : nVectors2 - iBlock2 * blockSize;
+            DAAL_INT startRow2     = iBlock2 * blockSize;
+            DAAL_INT endRow2       = startRow2 + nRowsInBlock2;
+
+            KernelRBFTask<algorithmFPType, cpu> * tlsLocal = tslTask.local();
 
-        daal::threader_for(blockNum2, blockNum2, [&](const size_t iBlock) {
-            const size_t numRowsInBlock = (iBlock != blockNum2 - 1) ? blockSize : nVectors2 - iBlock * blockSize;
-            const size_t startRow       = iBlock * blockSize;
-            const size_t finishRow      = startRow + numRowsInBlock;
+            algorithmFPType * mklBuff   = tlsLocal->mklBuff;
+            algorithmFPType * sqrDataA2 = tlsLocal->sqrDataA2;
 
-            ReadRows<algorithmFPType, cpu> mtA2(*const_cast<NumericTable *>(a2), startRow, numRowsInBlock);
+            ReadRows<algorithmFPType, cpu> mtA2(*const_cast<NumericTable *>(a2), startRow2, nRowsInBlock2);
             DAAL_CHECK_BLOCK_STATUS_THR(mtA2);
             const algorithmFPType * dataA2 = const_cast<algorithmFPType *>(mtA2.get());
 
-            for (size_t i = startRow, ii = 0; i < finishRow; ++i, ++ii)
+            for (size_t i = 0; i < nRowsInBlock2; ++i)
```

Comment:
would be `gemv` with displaces suitable here?

Reply:
This operation does not fit. `A[NxP] (?) A[NxP] = B[Nx1]`